### PR TITLE
 login and register modal

### PR DESCRIPTION
**Pull request by annaBalabak**

### CODE REVIEW

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                  onClick={handleResetPassword}
+                  type="submit"
+                >
+                  Забули пароль?
+                </button>
+              </div>
+              {resetPasswordClicked &&
+                resetPassword &&
+                email &&
+                validateEmail(email) && (
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:220]: Better to move it to a variable for readability

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+import { useNavigate } from 'react-router-dom';
+
+const LoginModal = ({ toggleLogin, toggleRegister }) => {
+  const navigate = useNavigate();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [submitted, setSubmitted] = useState(false);
+  const [resetPassword, setResetPassword] = useState(false);
+  const [resetPasswordClicked, setResetPasswordClicked] = useState(false);
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:19]: Do we need this variable?

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  const [loginSuccess, setLoginSuccess] = useState(false);
+  const [emailError, setEmailError] = useState('');
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:21]: Do we need this? you just set it to ' ' but never to real value

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    id="password"
+                    name="password"
+                    className={styles.dataInput}
+                    type="password"
+                    placeholder="Введіть ваш пароль"
+                    value={password}
+                    onChange={(e) => setPassword(e.target.value)}
+                  />
+                  {submitted && !password && (
+                    <p className={styles.errorMessage}>Введіть пароль</p>
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:178]: you can check it inside handeSubmit function and then set it to the variable passwordError

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+                    onChange={(e) => {
+                      setEmail(e.target.value.trim());
+                      setEmailError('');
+                    }}
+                  />
+                  {submitted && emailError && (
+                    <p className={styles.errorMessage}>{emailError}</p>
+                  )}
+                  {submitted && !email && (
+                    <p className={styles.errorMessage}>
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:139]: same here. check for all error cases inside handleSubmit function and set it to emailError or resetEmailError. But keep just 1 variable, no need for both of them

File: frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                    type="email"
+                    placeholder="Введіть електронну пошту"
+                    value={email}
+                    onChange={(e) => {
+                      setEmail(e.target.value.trim());
+                      setEmailExistsError('');
+                    }}
+                  />
+                  {submitted && emailExistsError && (
+                    <p className={styles.errorMessage}>{emailExistsError}</p>
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx:111]: you can create one variables for all errors and check them inside handleSubmit function.

File: frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                    placeholder="Введіть ваш пароль"
+                    value={password}
+                    onChange={(e) => setPassword(e.target.value)}
+                  />
+                  {submitted && !password && (
+                    <p className={styles.errorMessage}>Введіть пароль</p>
+                  )}
+                  {submitted && password && !validatePassword(password) && (
+                    <p className={styles.errorMessage}>
+                      Пароль має бути не менше 8 символів, містити латинські
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx:144]: Did we agree to use popup window for this info? like title attribute

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx (hunk 1 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:220]: done

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx (hunk 4 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:178]: Done

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx (hunk 5 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:139]: done

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx (hunk 3 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:21]: yes, we need it

File: frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx (hunk 2 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx:19]: yes

File: frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx (hunk 7 above)
Comment (annaBalabak) [reply frontend/src/components/common/LoginBtn/RegisterModal/RegisterModal.jsx:144]: The design is not ready yet

File: frontend/src/components/common/LoginBtn/ConfirmationModals/RegisterLetter.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        <div className={styles.mainContainer}>
+          <div className={styles.header}>
+            <p>Реєстрація</p>
+            <IconButton icon={<ICONS.closeMobile />} onClick={toggleModal} />
+          </div>
+
+          <p className={styles.email}>
+            {`На Вашу електронну пошту`} {}
+            <span className={styles.emailExample}> {userEmail} </span>
+            {}
```
Comment (halynarydasova) [inline frontend/src/components/common/LoginBtn/ConfirmationModals/RegisterLetter.jsx:20]: delete this

**Review by halynarydasova — COMMENTED** [review]
looks good for me

**Pull request merged**